src/test/java/io/opentracing/contrib/examples/request_context/CustomActiveSpanSource.java

```diff
@@ -0,0 +1,166 @@
+package io.opentracing.contrib.examples.request_context;
+
+import io.opentracing.ActiveSpan;
+import io.opentracing.ActiveSpanSource;
+import io.opentracing.Span;
+import io.opentracing.SpanContext;
+
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * If we create a wrapper for ActiveSpanSource then we won't able to obtain underlying Span inside
+ * ActiveSpanSource.activeSpan. That's why wrapper is insufficient, we need our own implementation :(
```

Comment:
This is probably a call for providing access to the underlying `Span`, i.e. `wrapped()`. In the latest PR from Ben that already happens, so we are likely going in the right direction ;)

About re-using (probably through wrapping, probably through subclassing) from `ThreadLocalActiveSpan` (or any other manager we provide later in the `utils` package), maybe we should also provide a design that lets users easily hook into it or override stuff, i.e. have `protected` methods that at the core operations to customize stuff.

Reply:
I'll try to implement this also for Ben's PR